### DISCUSSION TITLE
Using FloatingUIReact via UMD

### DISCUSSION THREAD

**Discussion by bgalartza** [floating-ui/floating-ui#2322] [Q&A]
Hello,
We would like to use the FloatingUIReact package via UMD, but we are having some trouble.

We loaded [this JS file](https://cdn.jsdelivr.net/npm/@floating-ui/react@0.24.0/dist/floating-ui.react.umd.js) from JSDeliver. Also the [Core](https://cdn.jsdelivr.net/npm/@floating-ui/core@1.2.6"></) and [Dom](https://cdn.jsdelivr.net/npm/@floating-ui/dom@1.2.7) packages, because apparently they are not included in the previous one.

When we try to use the `useFloating()` hook, we get the following error:
```
Uncaught InternalError: too much recursion
    React 9
        mountState
        useState
        useState
        useFloating
        useFloating
        useFloating
        useFloating
        useFloating
        useFloating
react-dom.inc.js:15652:16
    React 128
```
By using the browser JS debugger we can see that effectively the `useFloating` hook is calling itself.  You can see it in [this video](https://drive.google.com/file/d/1uoHIl_3QHZ8sCLdhw9y2MboPrE9h9iY-/view?usp=sharing)

Is this somehow expected?

The code we are using is the following (we are using it from Clojurescript)
```cljs
(defn- test
  []
  (let [floating (js/FloatingUIReactDOM.useFloating)
        refs (.-refs floating)
        style (.-floatingStyles floating)]
    [:div
     [:button
      {:ref (fn [el]
              (.setReference refs el))}
      "button"]
     [:div
      {:ref (fn [el]
              (.setFloating refs el))
       :style (js->clj style)}
      "tooltip"]]))
```

We don't have this problem if we load the FloatingUIReactDom library instead of FloatingUIReact. The very same code works as expected. But we are interested in using the second for the interactions.

Any hints what we might be doing wrong? Are those JS files expected to be used like that? 

Thank you

**Comment by atomiks** [comment]
This looks like a bug in the build system for UMD, as it's calling the package `FloatingUIReactDOM` not `FloatingUIReact`, which causes itself to call itself, rather than reference `@floating-ui/react-dom`'s exports. I think you must be the first one to try using it in this way as it's been broken the whole time 😆

**Reply by bgalartza** [reply]
Thank you for the fast reply. 
I have very little experience with the build tech stack that FloatingUI uses, but the bug might be in [this file](https://github.com/floating-ui/floating-ui/blob/master/packages/react/rollup.config.js#L28)? Specifically in lines 28 and 45. I would say that should be `FloatingUIReact` instead of `FloatingUIReactDom?`. I did the change, recompiled the project, and now it apparently works.

**Reply by atomiks** [reply]
Yes those two lines should remove the `DOM` suffix and it'll work - simple change

**Reply by bgalartza** [reply]
Should I submit a PR?

**Reply by atomiks** [reply]
Yeah feel free to
